src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -297,7 +330,11 @@ object ScalafixPlugin extends AutoPlugin {
     val parser = Def.setting(
       new ScalafixCompletions(
         workingDirectory = (ThisBuild / baseDirectory).value.toPath,
-        loadedRules = () => scalafixInterfaceProvider.value().availableRules(),
+        loadedRules = () =>
+          scalafixInterfaceProvider
+            // sbt Credentials is a task, so unfortunately it cannot be showed up here
+            .value((ThisBuild / scalafixResolvers).value)
```

Comment:
Can we do a tricky thing like separate the resolver from sbt and scalafix, make only resolvers that came from scalafix be used as cache key, this way it will make all tasks use the same cached instance, is that acceptable?

Reply:
Confirmed locally that parser is lazy. Also, there will be cache miss only if the build has sbt resolvers that are not in the `scalafixResolvers`, so effectively, for the vast majority of users which don't redefine `resolvers`, this won't change anything in terms of memory footprint since the `ScalafixInterface` will be the same for the parser and the actualy execution.